pkg/dequeuer/async_handler.go

```diff
@@ -170,12 +179,13 @@ func (h *AsyncMessageHandler) deletePayload(requestID string) {
 	}
 }
 
-func (h *AsyncMessageHandler) submitRequest(payload *userPayload, requestID string) (interface{}, error) {
+func (h *AsyncMessageHandler) submitRequest(payload *userPayload, headers http.Header, requestID string) (interface{}, error) {
 	req, err := http.NewRequest(http.MethodPost, h.config.TargetURL, payload.Body)
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
 
+	req.Header = headers
 	req.Header.Set("Content-Type", payload.ContentType)
```

Comment:
This one would be dropped in either (when relying on the headers file alone or if we no longer enforce it) case I think.